src/microtar.c

```diff
@@ -97,19 +111,21 @@ static int raw_to_header(mtar_header_t *h, const mtar_raw_header_t *rh) {
 
   /* Build and compare checksum */
   chksum1 = checksum(rh);
-  sscanf(rh->checksum, "%o", &chksum2);
+  sscanf(rh->checksum, "%8o", &chksum2);
   if (chksum1 != chksum2) {
     return MTAR_EBADCHKSUM;
   }
 
   /* Load raw header into header */
-  sscanf(rh->mode, "%o", &h->mode);
-  sscanf(rh->owner, "%o", &h->owner);
-  sscanf(rh->size, "%o", &h->size);
-  sscanf(rh->mtime, "%o", &h->mtime);
+  sscanf(rh->mode, "%8o", &h->mode);
```

Comment:
You don't need to specify octal size for `h->mode`, `h->owner`, and `h->mtime`. They're fixed values and guaranteed will not "overflow" from it's range.